src/Hive/Controllers/UserController.cs

```diff
@@ -67,6 +82,57 @@ public UserController(HiveContext context, IProxyAuthenticationService authServi
             permissions = perms;
         }
 
+        /// <summary>
+        /// Returns the user information for a requested user.
+        /// This is a JSON serialized collection of arbitrary objects, where the only two Hive defaults are the <see cref="User.Username"/> and <see cref="User.AlternativeId"/>.
+        /// Everything else can be added (or removed) as per plugins.
+        /// </summary>
+        /// <param name="username">The username to search for.</param>
+        /// <returns>The data to serialize.</returns>
+        [Route("info")]
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [SuppressMessage("Design", "CA1031:Do not catch general exception types", Justification = "Return expression does this")]
+        public async Task<ActionResult<Dictionary<string, object>>> GetUserInfo([FromQuery] string? username)
+        {
+            User? user;
+            if (string.IsNullOrWhiteSpace(username))
+            {
+                // User should be self
+                user = await HttpContext.GetHiveUser(authService).ConfigureAwait(false);
+            }
+            else
+            {
+                username = Uri.UnescapeDataString(username);
+                user = await authService.GetUser(username).ConfigureAwait(false);
+            }
+            // User must exist
+            if (user is null)
+            {
+                return NotFound();
+            }
+            try
+            {
+                // Permissions check
+                if (!permissions.CanDo(GetUserInfoActionName, new PermissionContext { User = await HttpContext.GetHiveUser(authService).ConfigureAwait(false), Username = username }, ref getUserParseState))
+                {
+                    return Unauthorized();
+                }
+                // Data to be serialized, pass through plugin first
+                // Start by adding username and sub in case (for some reason) the plugin wants to remove those
+                Dictionary<string, object> data = new() { { "username", username ?? user.Username }, { "sub", user.AlternativeId } };
+                plugin.Instance.ExposeUserInfo(data, user.AdditionalData);
+                return Ok(data);
+            }
+            catch
+            {
+                // If we have any type of error, for any reason, lets just go with BadRequest
+                return BadRequest();
```

Comment:
Are we certain we don't want to somehow propagate errors back to the user? It may make sense to return it in a sane API response, so that the user knows something went wrong. We could also just not catch and let ASP return a 500 for us.

Reply:
Yeah, this sounds reasonable to me. I have opted to just throw and see what happens, though having application crashes would obviously be less than ideal, so we may want to revisit this eventually.